riscv-rt/src/lib.rs

```diff
@@ -532,6 +532,26 @@
 #![no_std]
 #![deny(missing_docs)]
 
+/// Parse cfg attributes inside a global_asm call.
+#[allow(unused_macros)]
+macro_rules! cfg_global_asm {
+    {@inner, [$($x:tt)*], } => {
+        core::arch::global_asm!{$($x)*}
```

Comment:
Note: I moved this from `src/asm.rs` and did 2 changes:
- Fully qualify `core::arch::global_asm!()`
- All `unused_macros` since it's only used on `riscv32`, `riscv64`, or without `no-interrupts`.